ui/app/peers/create/configuration/configForm.tsx

```diff
@@ -0,0 +1,40 @@
+'use client';
+import { Label } from '@/lib/Label';
+import { RowWithTextField } from '@/lib/Layout';
+import { TextField } from '@/lib/TextField';
+import { PeerSetter } from './types';
+
+interface Setting {
+  label: string;
+  stateHandler: (value: string, setter: PeerSetter) => void;
+  type?: string;
+}
+
+interface ConfigProps {
+  settings: Setting[];
+  setter: PeerSetter;
+}
+
+export default function PgConfig(props: ConfigProps) {
+  return (
+    <>
+      {props.settings.map((setting, id) => {
+        return (
```

Comment:
I think making the form config-driven is a good idea. But we maybe need to revisit things like how default values can be set. For Example port default value is 5432 in config state variable, but we are not using it anywhere. perhaps a crude way to do it could be as shown below by adding a default value
 
```

export default function PgConfig(props: ConfigProps) {
  return (
    <>
      {props.settings.map((setting, id) => {
        let _defaultValue = setting.type === "number" && setting.label === "Port" ? "5432" : "";
        return (
          <RowWithTextField
            key={id}
            label={<Label as='label'>{setting.label}</Label>}
            action={
              <TextField
                variant='simple'
                type={setting.type}
                defaultValue={_defaultValue}
                onChange={(e) =>
                  setting.stateHandler(e.target.value, props.setter)
                }
              />
            }
          />
        );
      })}
    </>
  );
```

![image](https://github.com/PeerDB-io/peerdb/assets/39798643/33eb53b4-9c37-4fe2-ad0f-3049ff53deea)

Reply:
The default values are not really meant to be used in UI, but rather because protobuf-grpc doesn't seem to allow undefined values so we need dummies.

Reply:
Also thanks for the catch: the config value for the state here should be made generic to peer. fixed.